content/posts/drop-support-for-subversion-mercurial-bazaar.md

```diff
@@ -0,0 +1,72 @@
+title: Drop support for Subversion, Mercurial and Bazaar
+date: 2024-02-19
+category: Changelog
+tags: builders, deprecation
+authors: Manuel Kaufmann
+status: published
+
+
+We are announcing the dropping of support for Bazaar, Mercurial and Subversion version system controls starting on **Monday, June 3rd, 2024**.
+This means we will only support Git to do the initial clone and build your project's documentation.
+
+We've made this hard decision because [99% of our users](https://github.com/readthedocs/readthedocs.org/issues/8840) use Git
+and we can't cover the maintenance cost we were spending on Bazaar, Mercurial and Subversion for a handful amount of projects.
+Besides, those VCSs are not feature-complete in our platform compared to Git and the service provided is degraded for them.
+
+## Deprecation timeline
+
+We understand this change will affect some of our users, so we have a timeline to communicate this deprecation to our users effectively.
+
+* **Monday, April 1, 2024**: Do the first brownout (temporarily enforce this deprecation) for 12 hours: 00:01 PST to 11:59 PST (noon)
+* **Monday, May 6, 2024**: Do a second brownout (temporarily enforce this deprecation) for 24 hours: 00:01 PST to 23:59 PST (midnight)
+* **Monday, May 20, 2024**: Do a third and final brownout (temporarily enforce this deprecation) for 48 hours: 00:01 PST to May 21, 2024 23:59 PST (midnight)
+* **Monday, June 3, 2024**: Fully remove support for building documentation using “build.image” on the configuration file
+
+
+## Possible workarounds
+
+Unfortunately, there is no direct workaround on the Read the Docs side that you can follow to keep building your documentation using these VCSs.
+However, we consider one of the following options can be worth to consider:
+
+
+### Convert your repository to Git
+
+You could convert your Subversion or Mercurial repository to Git and then import the Git repository into Read the Docs to continue building your docs.
+GitHub has some guides to help you in this process:
+
+  * [Import your Subversion repository](https://docs.github.com/en/migrations/importing-source-code/using-the-command-line-to-import-source-code/importing-a-subversion-repository)
+  * [Import your Mercurial repository](https://docs.github.com/en/migrations/importing-source-code/using-the-command-line-to-import-source-code/importing-a-mercurial-repository)
+
+
+### Create a simple Git repository only clone the original one
```

Comment:
"Simple" :wink: 

This is a clever option, but I do hesitate to point users toward this. The configuration part is straight forward, but I feel like this might get very not simple quite quick for the user too.

At very least, I feel like we need more caveats on this because usage will all be very manual. On top of webhooks, how does the user manage versions in this workflow, or use pull request builds? This also uses build commands, which is another big jump for projects. There is probably more here.

I guess this should really point out that many of our features won't work and this is much more manual than built in Git support.

Reply:
Yeah, it's not simple, but small the word I wanted here.

I don't want to give them pretty detailed instructions on "how to perform the migration to keep all the exact features they currently have". The post is about a feature that's deprecated and there is no workaround for it. These are "ideas the can explore by themselves" --but won't give them what they currently have.

How I can write this in a clear way here?

Reply:
So, if we want to keep this example, I would frame this as a last resort and would explain that many features don't work.

I agree we shouldn't tell the user how to resolve all of the broken features. I was however pointing out the broken features as this section doesn't currently communicate exactly how much work there might be to this pattern.

That is, I would mention closer to:

> If you cannot use Git for your documentation repository, an intermediate Git repository could at least allow your project to build. Understand however that this is not a supported configuration and many of our features may not work or might require extra setup steps.

Also, could we do your example with build.jobs? That seems better than having to do everything manually with build.commands. At least with build.jobs, users could copy their configuration over to the new repository.

Reply:
I've done this in [`18606e8` (#264)](https://github.com/readthedocs/website/pull/264/commits/18606e87cdc7381695e8be01236cbd5b992cc203)